### PR TITLE
Performance test data

### PR DESCRIPTION
**Pull request by luontola**
#863

The most critical performance optimizations have been done. More will follow.

### CODE REVIEW

File: src/clj/rems/api/applications_v2.clj [hunk 1]
```diff
@@ -65,10 +67,20 @@
           :application/form
           :application/licenses))
 
+(mount/defstate all-applications-cache
+  :start (atom {:last-processed-event-id 0
+                :applications nil}))
```
Comment (opqdonut) [inline src/clj/rems/api/applications_v2.clj]: core.cache has nice functional caches with various startegies, please use it instead

File: src/clj/rems/db/test_data.clj [hunk 2]
```diff
@@ -628,6 +631,91 @@
...
+                                             :organization "perf"
+                                             :licenses [license-id]}
+                                            owner)
+                                  _ (assert (:success resource))
+                                  cat-item (catalogue/create-catalogue-item! {:title (str "Performance test resource " uuid)
+                                                                              :form form-id
+                                                                              :resid (:id resource)
+                                                                              :wfid workflow-id})]
+                              (:id cat-item))))
+        user-ids (vec (for [n (map inc (range user-count))]
```
Comment (opqdonut) [inline src/clj/rems/db/test_data.clj]: could also use `(range 1 (inc user-count))`

File: src/clj/rems/db/test_data.clj [hunk 3]
```diff
@@ -628,6 +631,91 @@
...
...c eu molestie eros. Nunc iaculis rhoncus enim, vel mattis felis fringilla condimentum. Interdum et malesuada fames ac ante ipsum primis in faucibus. Aenean ac augue nulla. Phasellus vitae nulla lobortis, mattis magna ac, gravida ipsum. Aenean ornare non nunc non luctus. Aenean lacinia lectus nec velit finibus egestas vel ut ipsum. Cras hendrerit rhoncus erat, vel maximus nunc.\n\nPraesent quis imperdiet quam. Praesent ligula tellus, consectetur sed lacus eu, malesuada condimentum tellus. Donec et diam hendrerit, dictum diam quis, aliquet purus. Suspendisse pulvinar neque at efficitur iaculis. Nulla erat orci, euismod id velit sed, dictum hendrerit arcu. Nulla aliquam molestie aliquam. Duis et semper nisi, eget commodo arcu. Praesent rhoncus, nulla id sodales eleifend, ante ipsum pellentesque augue, id iaculis sem est vitae est. Phasellus cursus diam a lorem vestibulum sodales. Nullam lacinia tortor vel tellus commodo, sit amet sodales quam malesuada.\n\nNulla tempor lectus vel arcu feugiat, vel dapibus ex dapibus. Maecenas purus justo, aliquet et sem sit amet, tincidunt venenatis dui. Nulla eget purus id sapien elementum rutrum eu vel libero. Cras non accumsan justo posuere."}]})))
```
Comment (opqdonut) [inline src/clj/rems/db/test_data.clj]: I'd prefer a constant for this. We already have one on the cljs side, perhaps move it to cljc?

File: src/clj/rems/db/test_data.clj [hunk 4]
```diff
@@ -628,6 +631,91 @@
...
+                                                                              :resid (:id resource)
+                                                                              :wfid workflow-id})]
+                              (:id cat-item))))
+        user-ids (vec (for [n (map inc (range user-count))]
+                        (let [user-id (str "perftester" n)]
+                          (users/add-user! user-id {:eppn user-id
+                                                    :mail (str user-id "@example.com")
+                                                    :commonName (str "Performance Tester " n)})
+                          user-id)))]
+    (doseq [_ (range application-count)]
```
Comment (opqdonut) [inline src/clj/rems/db/test_data.clj]:
```
(dotimes [_ application-count]

File: src/clj/rems/db/test_data.clj (hunk 3 above)
Comment (luontola) [reply src/clj/rems/db/test_data.clj]: It's used for quite different purpose than the strings in the component guide. There the lipsum is for visual filler. Here it's for memory filler. That's why I prefer to keep them separate.

Comment (opqdonut) [reply src/clj/rems/db/test_data.clj]: fine by me

File: src/clj/rems/api/applications_v2.clj (hunk 1 above)
Comment (luontola) [reply src/clj/rems/api/applications_v2.clj]: The events are immutable and the injections are not yet cached in state, so for now an atom is enough. I'll keep core.cache in mind when doing further optimizations.

**Pull request merged**